### PR TITLE
Added connection to MySql

### PR DESCRIPTION
**Pull request by Barburon**

### CODE REVIEW

File: src/main/java/com/cursor/dao/interfaces/CRUD.java [hunk 1]
```diff
@@ -1,15 +1,16 @@
 package com.cursor.dao.interfaces;
 
+import java.sql.SQLException;
 import java.util.List;
 
 public interface CRUD<T> {
-    boolean create(T entity);
+    boolean create(T entity) throws SQLException;
```
Comment (lazermann111) [inline src/main/java/com/cursor/dao/interfaces/CRUD.java:7]: I propose to remove 'throws' clause cause not all childs of interface actually throwing it

File: src/main/java/com/cursor/service/DashboardImpl.java [hunk 2]
```diff
@@ -1,47 +1,54 @@
...
 import java.util.Comparator;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class DashboardImpl implements Dashboard {
-    private TicketDao tickets = TicketDao.getInstance();
-    private UserDao users = UserDao.getInstance();
+//    private TicketDaoInMem tickets = TicketDaoInMem.getInstance();
+//    private UserDaoInMem users = UserDaoInMem.getInstance();
+    private final TicketDaoDb tickets = TicketDaoDb.getInstance();
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/DashboardImpl.java:23]: left part of expression shoud be an interface: TicketDaoDb -> TicketDao

File: src/main/java/com/cursor/service/DashboardImpl.java [hunk 3]
```diff
@@ -1,47 +1,54 @@
...
+    private final UserDaoDb users = UserDaoDb.getInstance();
 
     @Override
-    public int getTotalTime(User user) {
+    public int getTotalTime(User user) throws SQLException {
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/DashboardImpl.java:27]: dashboard shouldnt be aware about SQLexceptions :)

File: src/main/java/com/cursor/service/TicketServiceImpl.java [hunk 4]
```diff
@@ -1,50 +1,53 @@
...
 import com.cursor.service.exceptions.NotFoundException;
 import com.cursor.service.interfaces.TicketService;
+
+import java.sql.SQLException;
 import java.util.List;
 
 public class TicketServiceImpl implements TicketService {
-
-    private final TicketDao tickets = TicketDao.getInstance();
+    private final TicketDaoDb tickets = TicketDaoDb.getInstance();
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/TicketServiceImpl.java:15]: TicketDaoDb -> TicketDao

File: src/main/java/com/cursor/service/UserServiceImpl.java [hunk 5]
```diff
@@ -24,30 +27,30 @@ public void registerUser(User user) {
     }
 
     @Override
-    public void loginUser(String username, String password) {
+    public void loginUser(String username, String password) throws SQLException {
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/UserServiceImpl.java:30]: service layer shouldnt be aware of SQLException

File: src/main/java/com/cursor/service/UserServiceImpl.java [hunk 6]
```diff
@@ -24,30 +27,30 @@ public void registerUser(User user) {
...
         checkExistence(username, password);
     }
 
     @Override
-    public void create(User entity) {
+    public void create(User entity) throws SQLException {
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/UserServiceImpl.java:35]: service layer shouldnt be aware of SQLException

File: src/main/java/com/cursor/service/interfaces/Dashboard.java [hunk 7]
```diff
@@ -5,24 +5,25 @@
 import com.cursor.model.enums.Priority;
 import com.cursor.model.enums.Status;
 
+import java.sql.SQLException;
 import java.util.List;
 
 public interface Dashboard {
 
-    int getTotalTime(User user);
+    int getTotalTime(User user) throws SQLException;
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/interfaces/Dashboard.java:13]: service layer shouldnt be aware of SQLException

File: src/main/java/com/cursor/view/Actions.java [hunk 8]
```diff
@@ -22,7 +23,7 @@
     UserServiceImpl userService = new UserServiceImpl();
     Dashboard dashboard = new DashboardImpl();
 
-    public void showActionsMenu() {
+    public void showActionsMenu() throws SQLException {
```
Comment (lazermann111) [inline src/main/java/com/cursor/view/Actions.java:26]: view layer  shouldnt be aware of SQLException

**Review by lazermann111 — CHANGES_REQUESTED** [review]
plz see commetns

**Review by lazermann111 — APPROVED** [review]

**Review by AmorBielyi — APPROVED** [review]

**Review by Julia1910 — APPROVED** [review]

**Pull request merged**